sdk/go/common/workspace/plugins_test.go

```diff
@@ -541,6 +541,32 @@ func TestPluginDownload(t *testing.T) {
 			assert.Equal(t, expectedBytes, readBytes)
 		})
 	})
+	t.Run("Test Downloading From GitLab Releases", func(t *testing.T) {
+		t.Setenv("GITLAB_TOKEN", token)
+		version := semver.MustParse("1.23.4")
+		spec := PluginSpec{
+			PluginDownloadURL: "gitlab://gitlab.com/278964",
+			Name:              "mock-gitlab",
+			Version:           &version,
+			Kind:              PluginKind("resource"),
+		}
+		source, err := spec.GetSource()
+		assert.NoError(t, err)
+		getHTTPResponse := func(req *http.Request) (io.ReadCloser, int64, error) {
+			assert.Equal(t,
+				"https://gitlab.com/api/v4/projects/278964/releases/v1.23.4/downloads/"+
+					"pulumi-resource-mock-gitlab-v1.23.4-windows-arm64.tar.gz", req.URL.String())
+			assert.Equal(t, fmt.Sprintf("Bearer %s", token), req.Header.Get("Authorization"))
```

Comment:
optional:
It's nicer if we can keep the test self-contained where possible.
There doesn't appear to be anything preventing use of a local variable/value for the token here.
That'll also allow this assertion to be free of the sprintf, and just assert an exact string:

```go
t.Setenv("GITLAB_TOKEN", "mysecrettoken")
// ...
assert.Equal(t, "Bearer mysecrettoken", req.Header.Get("Authorization"))
```

Reply:
Maybe, we could do a pass over all of them I suppose but you need to start picking different token values or golint starts complaining about seeing the same string over and over, and then peppering the code with lint-off comments doesn't look much better than what's already there.